compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -499,12 +499,15 @@ class ComposeScene internal constructor(
      */
     val contentSize: IntSize
         get() {
-            check(!isClosed) { "ComposeScene is closed" }
             val mainOwner = mainOwner ?: return IntSize.Zero
-            mainOwner.measureAndLayout()
-            return mainOwner.contentSize
+            return contentSizeInConstraints(mainOwner.constraints)
```

Comment:
We should cache this or turn it into a function. Code that previously assumed this was a quick operation will need to be rewritten. For example in `SemanticsNode.isInScreenBounds()`:

```
    return nodeBoundsInWindow.top >= 0 &&
        nodeBoundsInWindow.left >= 0 &&
        nodeBoundsInWindow.right <= composeView.contentSize.width &&
        nodeBoundsInWindow.bottom <= composeView.contentSize.height
```

Reply:
Actually, are you sure this isn't breaking anything? Before, contentSize was the actual size of the scene. Now it's the result of some computation - why is it guaranteed to match the real size of the scene?

Reply:
Ah, I missed this particular case, indeed this one can break, thank you

Reply:
Mentioned code was recently changed by Ivan. Also, this API always was not that cheap, since it did measure and layout.
So, nothing is changed here, code still works the same, the only difference is that if preferredSize is called and then contentSize, then measureAndLayout will take more time, since constraints are changed